addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/dialogs/ZestClientWindowResizeDialog.java

```diff
@@ -0,0 +1,115 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.zest.dialogs;
+
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.script.ScriptNode;
+import org.zaproxy.zap.extension.zest.ExtensionZest;
+import org.zaproxy.zap.extension.zest.ZestScriptWrapper;
+import org.zaproxy.zap.view.StandardFieldsDialog;
+import org.zaproxy.zest.core.v1.ZestClientWindowResize;
+import org.zaproxy.zest.core.v1.ZestStatement;
+
+@SuppressWarnings("serial")
+public class ZestClientWindowResizeDialog extends StandardFieldsDialog implements ZestDialog {
+
+    private static final String FIELD_WINDOW_HANDLE = "zest.dialog.client.label.windowHandle";
+    private static final String FIELD_WIDTH = "zest.dialog.client.label.width";
+    private static final String FIELD_HEIGHT = "zest.dialog.client.label.height";
+
+    private static final long serialVersionUID = 1L;
+
+    private ExtensionZest extension = null;
+    private ScriptNode parent = null;
+    private ScriptNode child = null;
+    private ZestScriptWrapper script = null;
+    private ZestStatement request = null;
+    private ZestClientWindowResize client = null;
+    private boolean add = false;
+
+    public ZestClientWindowResizeDialog(ExtensionZest ext, Frame owner, Dimension dim) {
+        super(owner, "zest.dialog.clientWindowResize.add.title", dim);
+        this.extension = ext;
+    }
+
+    public void init(
+            ZestScriptWrapper script,
+            ScriptNode parent,
+            ScriptNode child,
+            ZestStatement req,
+            ZestClientWindowResize client,
+            boolean add) {
+        this.script = script;
+        this.add = add;
+        this.parent = parent;
+        this.child = child;
+        this.request = req;
+        this.client = client;
+
+        this.removeAllFields();
+
+        if (add) {
+            this.setTitle(Constant.messages.getString("zest.dialog.clientWindowResize.add.title"));
+        } else {
+            this.setTitle(Constant.messages.getString("zest.dialog.clientWindowResize.edit.title"));
+        }
+
+        // Pull down of all the valid window ids
+        List<String> windowIds = new ArrayList<>(script.getZestScript().getClientWindowHandles());
+        Collections.sort(windowIds);
+        this.addComboField(FIELD_WINDOW_HANDLE, windowIds, client.getWindowHandle());
+        this.addTextField(FIELD_WIDTH, String.valueOf(client.getX()));
+        this.addTextField(FIELD_HEIGHT, String.valueOf(client.getY()));
+    }
+
+    @Override
+    public void save() {
+        client.setWindowHandle(this.getStringValue(FIELD_WINDOW_HANDLE));
+        client.setX(Integer.valueOf(this.getStringValue(FIELD_WIDTH)));
+        client.setY(Integer.valueOf(this.getStringValue(FIELD_HEIGHT)));
```

Comment:
This leads to exceptions if the user sets other thing than an integer, ideally the UI should prevent setting invalid values (other dialogues use a number spinner), otherwise it should validate the fields. Same applies to scroll.

Reply:
using a spinner will be a very slow process to set values. Is there any other alternative for this ?

Reply:
The number spinner allows to set values directly.

Reply:
okayy